vault/logical_system_activation_flags.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const (
+	EnvHVDTierSecretsSync = "VAULT_HVD_ADMIN_SECRETS_SYNC"
+
+	paramFeatureName = "feature_name"
+	descFeatureName  = "The name of the feature to be activated."
+	summaryList      = "Returns the available and activated activation-flagged features."
+	summaryUpdate    = "Activate a flagged feature."
+
+	prefixActivationFlags         = "activation-flags"
+	verbActivationFlagsActivate   = "activate"
+	verbActivationFlagsDeactivate = "deactivate"
+
+	fieldActivated   = "activated"
+	fieldUnactivated = "unactivated"
+
+	helpSynopsis    = "Returns information about Vault's features that require a one-time activation step."
+	helpDescription = `
+This path responds to the following HTTP methods.
+	GET /
+		Returns the available and activated activation-flags.
+
+	PUT|POST /<feature-name>/activate
+		Activates the specified feature. Cannot be undone.`
+)
+
+// Register CRUD functions dynamically.
```

Comment:
I think it would help to describe the expectations around safe usage of these variables. For example I guess we should only ever mutate them during `init` or server construction code right? It's not safe to just swap these out once Vault core is running.